src/main/java/org/pac4j/vertx/handler/impl/CallbackHandler.java

```diff
@@ -34,75 +40,67 @@
  * @since 1.0.0
  *
  */
-public class CallbackHandler extends BasePac4JAuthHandler {
-
-  protected static final Logger logger = LoggerFactory.getLogger(CallbackHandler.class);
+public class CallbackHandler implements Handler<RoutingContext> {
 
-    public CallbackHandler(final Pac4jWrapper wrapper, final Pac4jAuthProvider authProvider, final Pac4jAuthHandlerOptions options) {
-        super(wrapper, authProvider, options);
-    }
+  protected static final Logger LOG = LoggerFactory.getLogger(CallbackHandler.class);
 
-  /**
-   * Authenticates the given request.
-   *
-   * @param routingContext the vertx-web routing context for the request
-   * @param sessionAttributes the session attributes
-   * @param handler the handler
-   */
-  protected void authenticate(final RoutingContext routingContext, final Pac4jSessionAttributes sessionAttributes,
-                              final Handler<Pac4jAuthenticationResponse> handler) {
+  private final HttpActionHandler httpActionHandler = new DefaultHttpActionHandler();
+  private final Vertx vertx;
+  private final Config config;
 
-    wrapper.authenticate(routingContext, clientName, sessionAttributes, handler);
+  protected String defaultUrl = Pac4jConstants.DEFAULT_URL_VALUE;
 
+  public CallbackHandler(final Vertx vertx,
+                           final Config config) {
+      this.vertx = vertx;
+      this.config = config;
   }
 
   @Override
-  protected void retrieveUserProfile(final RoutingContext routingContext, final Pac4jSessionAttributes sessionAttributes,
-                                     final Handler<Pac4jAuthenticationResponse> handler) {
-    authenticate(routingContext, sessionAttributes, handler);
-  }
-
-  @Override
-    protected void authenticationSuccess(final RoutingContext routingContext, final Pac4jSessionAttributes sessionAttributes) {
-
-        redirectToTarget(routingContext, sessionAttributes);
-    }
-
-    protected void authenticationFailure(final RoutingContext routingContext, final Pac4jSessionAttributes sessionAttributes) {
-
-        redirectToTarget(routingContext, sessionAttributes);
-    }
-
-    private void redirectToTarget(final RoutingContext routingContext, final Pac4jSessionAttributes sessionAttributes) {
-
-      String requestedUrl = retrieveOriginalUrl(sessionAttributes);
-      saveUrl(null, sessionAttributes);
-      final String redirectUrl = defaultUrl(requestedUrl, BaseConfig.getDefaultSuccessUrl());
-      pac4jAuthProvider.saveSessionAttributes(routingContext, sessionAttributes);
-      HttpResponseHelper.redirect(routingContext.request(), redirectUrl);
-
+  public void handle(RoutingContext event) {
+
+    // Can we complete the authentication process here?
+    final VertxWebContext webContext = new VertxWebContext(event);
+    final ProfileManager profileManager = new ProfileManager(webContext);
+    final Client client = config.getClients().findClient(webContext);
```

Comment:
I make your life hard, but the source code has slightly changed: there will be a check to ensure the client is an `IndirectClient` one: https://github.com/pac4j/j2e-pac4j/pull/20/files#diff-dd8336ff9f7458dadab996b4d74c7cb8R74